modules/raft/tech-notes/src/primaryReplica.puml

```diff
@@ -0,0 +1,34 @@
+@startuml primaryReplica
+title PrimaryReplica and Replication Group communication (diagram 1)
+
+skinparam maxMessageSize 400
+skinparam defaultFontSize 12
+
+participant PlacementDriver
+participant PrimaryReplica
+
+participant Replica1 [
+Replica1
+Leader for term 1
+]
+
+participant Replica2 [
+Replica2
+Leader for term 2
+]
+
+PrimaryReplica -> Replica1 : Send a changePeersAsync request (node is the leader at the moment)
+Replica1 -> Replica1 : Leader was stepped down and new leader election will start.
+Replica2 -> Replica2 : Current node elected as a leader.
+Replica2 -> PrimaryReplica : Send a message about the new leader elected. [see 3.1 details]
+PrimaryReplica -> PrimaryReplica : Check local state for ongoing rebalance for the replica group.
```

Comment:
Actually, replica1 should transfer leader to primary replica, to make the leader colocated with primary, instead of just stepping down. But as far as i can see, this diagram is correct in terms of rebalancing and maybe describes the flow better. It's up to you to change anything here.

Reply:
Yes, you are right. But fo this case it doesn't matter - will it be collocated or not, just for the explicitness of digram, I think we can leave it as is